Statistics/Distributions/LogNormal.cs

```diff
@@ -9,121 +9,111 @@ namespace Statistics.Distributions
-            StandardDeviation = standardDeviation;
+            StandardDeviation = sd;
             SampleSize = sampleSize;
-            Min = double.NegativeInfinity;
-            Max = double.PositiveInfinity;
-            BuildFromProperties();
+            _ProbabilityRange = IRangeFactory.Factory(0.0, 1.0);
+            Min = InverseCDF(0.0000000000001);
+            Max = InverseCDF(1-0.0000000000001);
+            State = Validate(new Validation.LogNormalValidator(), out IEnumerable<Utilities.IMessage> msgs);
+            Messages = msgs;
+
         }
-        public LogNormal(double mean, double standardDeviation, double min, double max, int sampleSize = int.MaxValue)
+        public LogNormal(double mean, double sd, double minValue, double maxValue, int sampleSize = int.MaxValue)
         {
             Mean = mean;
-            StandardDeviation = standardDeviation;
+            StandardDeviation = sd;
             SampleSize = sampleSize;
-            Min = min;
-            Max = max;
+            Min = minValue;
+            Max = maxValue;
             Truncated = true;
-            BuildFromProperties();
-            
-        }
-        public void BuildFromProperties()
-        {
-
-            if (!Validation.LogNormalValidator.IsConstructable(Mean, StandardDeviation, SampleSize, out string msg)) throw new InvalidConstructorArgumentsException(msg);
-            _Distribution = new MathNet.Numerics.Distributions.LogNormal(Mean, StandardDeviation);
             _ProbabilityRange = FiniteRange(Min, Max);
-            Range = IRangeFactory.Factory(_Distribution.InverseCumulativeDistribution(_ProbabilityRange.Min), _Distribution.InverseCumulativeDistribution(_ProbabilityRange.Max));
-            Min = Range.Min;
-            Max = Range.Max;
             State = Validate(new Validation.LogNormalValidator(), out IEnumerable<Utilities.IMessage> msgs);
             Messages = msgs;
-            _Constructed = true;
+            
         }
+
         #endregion
 
         #region Functions
         public IMessageLevels Validate(Utilities.IValidator<LogNormal> validator, out IEnumerable<Utilities.IMessage> msgs)
         {
             return validator.IsValid(this, out msgs);
         }
-        private IRange<double> FiniteRange(double min = double.NegativeInfinity, double max = double.PositiveInfinity)
+        private IRange<double> FiniteRange(double min, double max)
         {
-            double pmin = 0, epsilon = 1 / 1000000000d;
+            double pmin = 0;
             double pmax = 1 - pmin;
-            if (min.IsFinite() || max.IsFinite())//not entirely sure how inclusive or works with one sided truncation and the while loop below.
-            {
-                pmin = _Distribution.CumulativeDistribution(min);
-                pmax = _Distribution.CumulativeDistribution(max);
-            }
-            while (!(min.IsFinite() && max.IsFinite()))
+            if (min.IsFinite() || max.IsFinite())
```

Comment:
what happens if min or max is not finite? 

Should this be an "AND" rather than an "OR"?